File: aiterm-server/internal/api/terminal.go
package api

import (
	"encoding/json"
	"net/http"
	"time"
)

type terminalExecuteRequest struct {
	Command string `json:"command"`
	NodeID  string `json:"node_id"`
}

type terminalExecuteResponse struct {
	Command   string `json:"command"`
	Output    string `json:"output"`
	ExitCode  int    `json:"exit_code"`
	TimedOut  bool   `json:"timed_out"`
	NodeID    string `json:"node_id"`
	NodeName  string `json:"node_name"`
	Timestamp string `json:"timestamp"`
}

func handleTerminalExecute(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodPost {
		writeMethodNotAllowed(w)
		return
	}

	var req terminalExecuteRequest
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		writeJSON(w, http.StatusBadRequest, response{
			Code:    1000,
			Message: "invalid request",
		})
		return
	}

	if req.Command == "" {
		writeJSON(w, http.StatusBadRequest, response{
			Code:    1001,
			Message: "command is required",
		})
		return
	}

	nodeID := req.NodeID
	if nodeID == "" {
		nodeID = "1"
	}

	nodeName := "local"
	if node, ok := defaultStore.getNode(nodeID); ok {
		nodeName = node.Name
	}

	result := executeCommand(r.Context(), req.Command)

	output := ""
	for _, line := range result.Lines {
		output += line.Content + "\n"
	}

	writeJSON(w, http.StatusOK, response{
		Code:    0,
		Message: "ok",
		Data: terminalExecuteResponse{
			Command:   req.Command,
			Output:    output,
			ExitCode:  result.ExitCode,
			TimedOut:  result.TimedOut,
			NodeID:    nodeID,
			NodeName:  nodeName,
			Timestamp: time.Now().UTC().Format(time.RFC3339),
		},
	})
}

File: aiterm-server-python/app/api/v1/terminal.py
from fastapi import APIRouter, Depends
from datetime import datetime

from app.models import Response, TerminalExecuteRequest, TerminalExecuteResponse
from app.services import execute_command, NodeService
from app.api.deps import get_node_service
from app.utils import now_iso

router = APIRouter(prefix="/terminal", tags=["terminal"])


@router.post("/execute")
async def execute_terminal_command(
    request: TerminalExecuteRequest,
    node_service: NodeService = Depends(get_node_service)
):
    if not request.command:
        return Response(code=1001, message="command is required")

    node_id = request.node_id or "1"
    node_name = "local"

    node = await node_service.get_node(node_id)
    if node:
        node_name = node.name

    result = await execute_command(request.command)

    output = "\n".join(content for _, content in result.lines)

    return Response(
        data=TerminalExecuteResponse(
            command=request.command,
            output=output,
            exit_code=result.exit_code,
            timed_out=result.timed_out,
            node_id=node_id,
            node_name=node_name,
            timestamp=now_iso()
        ).model_dump()
    )
